src/cluster.c

```diff
@@ -6453,6 +6453,85 @@ void removeChannelsInSlot(unsigned int slot) {
     zfree(channels);
 }
 
+/* The SCAN command completely relies on scanGenericCommand. */
+
+/* Returns a unique 3 character hash key. */
+static char HashKeyTable[CLUSTER_SLOTS][4] = {0};
+
+static void fillHashKeys(void) {
+    int remaining = CLUSTER_SLOTS;
+    const char *valid_chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
```

Comment:
I will experiment later, but: I wonder if there exists a 32-character subset of the ?95? printable ASCII for which 3 chars covers all slots. Because if there is, the 5 bits each can be packed into 2 bytes for storage. A small saving, but tempting to try the 95C32 combinations to find out!

Reply:
Well, I burned a CPU-week trying various combinations without luck (and yes I do understand the size of the problem space). If nothing else, it helped check system stability on a new CPU. No joy, though. Maybe use 3 bytes ASCII or 2 bytes binary-safe :)